prime-router/docs/proposals/0017-metadata-tables-cleanup.md

```diff
@@ -0,0 +1,441 @@
+    'The state ID for the provider';
+comment on column provider.provider_country_id is
+    'The country ID for the provider';
+comment on column provider.created_at is 'The timestamp for the record creation';
+comment on column provider.last_modified_at is 'The timestamp for the record modification';
+
+
+-- the sender of the message
+create table sender
+(
+    sender_id                           bigserial not null primary key
+    , sender_name                       varchar(512) not null
+    , organization_name                 varchar(512) not null
+    , sender_oid                        varchar(128) null
+    , sender_namespace_id               varchar(128) null
+    , created_at                        timestamp default current_timestamp
+    , last_modified_at                  timestamp default current_timestamp
+);
+
+comment on table sender is
+    'The sender of an ELR message.';
+
+-- our core ELR message table.
+create table elr_message
+(
+    elr_message_id                      bigserial not null primary key
+    , topic                             varchar(128) not null
+    , report_id                         uuid not null
+    , tracking_id                       varchar(512)
+    , report_index                      integer not null
+    , internal_message_id               uuid null
+    , previous_message_id               varchar(512) not null
+    , sender_id                         bigint not null references sender(sender_id)
+    , sending_application_name          varchar(128) null
+    , sending_application_id            varchar(128) null
+    , created_at                        timestamp not null default current_timestamp
+    -- patient info
+    , patient_state_id                  bigint null references state(state_id)
+    , patient_county_id                 bigint null references county(county_id)
+    , patient_country_id                bigint null references country(country_id)
+    , patient_postal_code               varchar(64) null
+    , patient_gender_code               varchar(64) null
+    , patient_gender                    varchar(128) null
+    , patient_ethnicity_code            varchar(64) null
+    , patient_ethnicity                 varchar(128) null
+    , patient_race_code                 varchar(64) null
+    , patient_race                      varchar(128) null
+    , patient_tribal_citizenship_code   varchar(64) null
+    , patient_tribal_citizenship        varchar(128) null
+    , patient_species_code              varchar(64) null
+    , patient_species                   varchar(128) null
+    , patient_age                       varchar(64) null
+);
+
+comment on table elr_message is
+    'Our main ELR message table. This represents a single ORU_R01 message, though the data does not need to necessarily come from HL7.';
+comment on column elr_message.elr_message_id is
+    'A sequential primary key for the message table';
+
+
+-- there can be many AOE questions for a test. let's catch them here
```

Comment:
Thought: If we have multiple OBRs in a single message and multiple of them have associated AOEs does this become an issue. I don't think so, but worth thinking about.